### PR TITLE
CORE-2289 add stdlib.contract

### PR DESCRIPTION
**Pull request by erer1243**

### CODE REVIEW

File: docs/src/frontend/index.md [hunk 1]
```diff
@@ -833,13 +833,16 @@ The third thing you should do in a frontend is create a contract handle, so you
...
 ```js
-acc.contract(bin, ?info) => ctc
+acc.contract(bin: Backend, info?: Promise<ContractInfo>) => Contract
+stdlib.contract(bin: Backend, info?: Promise<ContractInfo>) => Promise<Contract>
 ```
 
-Returns a Reach contract handle based on the `{!js} bin` argument provided with access to the account `{!js} acc`.
+Returns a Reach contract handle based on the `{!js} bin` argument, and optional `ContractInfo` provided.
+The `{!js} acc.contract` form will use the account `{!js} acc` to interact with the returned contract.
+The `{!js} stdlib.contract` form is equivalent to `{!js} (await stdlib.createAccount()).contract`, generating an account on demand.
```
Comment (jeapostrophe) [inline docs/src/frontend/index.md:842]:
```suggestion
The `{!js} stdlib.contract` form is equivalent to `{!js} (await stdlib.createAccount()).contract`, generating an account on demand.
(This means that the contract handle may not be used for non-read-only interaction.)
```

File: docs/src/frontend/index.md [hunk 2]
```diff
@@ -853,16 +856,16 @@ This deployment can only happen one time, so subsequent attempts will fail with
 This function may emit warnings if there is any danger, risk, or subtlety to using this contract on your chosen consensus network.
 You can omit this warning by setting `{!cmd} REACH_NO_WARN`, but we recommend that you do not.
 
-This function does not block.
+The `acc.contract` form of the function does not block.
```
Comment (jeapostrophe) [inline docs/src/frontend/index.md:859]:
```suggestion
The `{!js} acc.contract` form of the function does not block.
```

File: js/stdlib/ts/ALGO.ts [hunk 3]
```diff
@@ -2707,6 +2707,9 @@ const launchToken = async (accCreator: Account, name: string, sym: string, opts:
   return { name, sym, id, mint, optOut };
 }
 
+const contract = (bin: Backend, ctcInfo?: Promise<ContractInfo>): Promise<Contract> =>
+  createAccount().then(acc => acc.contract(bin, ctcInfo));
```
Comment (jeapostrophe) [inline js/stdlib/ts/ALGO.ts:2711]: I think there should be something in `shared_impl` to make it easy to do this and keep them in all in sync

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: js/stdlib/ts/ALGO.ts (hunk 3 above)
Comment (erer1243) [reply js/stdlib/ts/ALGO.ts]: i did this, but i don't think it's better

File: js/stdlib/ts/ALGO.ts [hunk 4]
```diff
@@ -2730,6 +2731,6 @@ const launchToken = async (accCreator: Account, name: string, sym: string, opts:
     parseCurrency, minimumBalance, formatCurrency,
     reachStdlib, algosdk,
     connector, standardUnit, atomicUnit,
-    tokensAccepted,
-  };
+    tokensAccepted
```
Comment (jeapostrophe) [inline js/stdlib/ts/ALGO.ts:2734]: Heresy!

**Review by jeapostrophe — APPROVED** [review]

File: js/stdlib/ts/ALGO.ts (hunk 4 above)
Comment (erer1243) [reply js/stdlib/ts/ALGO.ts:2734]: hearsay

**Pull request merged**